diff --git a/TMessagesProj/src/main/java/org/telegram/ui/SettingsActivity.java b/TMessagesProj/src/main/java/org/telegram/ui/SettingsActivity.java
--- a/TMessagesProj/src/main/java/org/telegram/ui/SettingsActivity.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/SettingsActivity.java
@@ -962,8 +962,11 @@
     @NonNull
     private WindowInsetsCompat onApplyWindowInsets(@NonNull View v, @NonNull WindowInsetsCompat insets) {
         final Insets systemInsets = AndroidUtilities.getDefaultWindowInsets(insets, false);
-        navigationBarHeight = systemInsets.bottom;
-        final int statusBarHeight = systemInsets.top;
+        navigationBarHeight = systemInsets.bottom;
+        additionNavigationBarHeight = hasMainTabs
+                ? dp(DialogsActivity.MAIN_TABS_HEIGHT_WITH_MARGINS)
+                : 0;
+        final int statusBarHeight = systemInsets.top;
         listView.setPadding(0, statusBarHeight + dp(12), 0, navigationBarHeight + additionNavigationBarHeight);
         return WindowInsetsCompat.CONSUMED;
     }
diff --git a/TMessagesProj/src/main/java/org/telegram/ui/ContactsActivity.java b/TMessagesProj/src/main/java/org/telegram/ui/ContactsActivity.java
--- a/TMessagesProj/src/main/java/org/telegram/ui/ContactsActivity.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/ContactsActivity.java
@@ -1583,9 +1583,15 @@
     @NonNull
     private WindowInsetsCompat onApplyWindowInsets(@NonNull View v, @NonNull WindowInsetsCompat insets) {
         final Insets systemInsets = AndroidUtilities.getDefaultWindowInsets(insets, false);
-        navigationBarHeight = systemInsets.bottom;
-
-        checkUi_listViewPadding();
+        navigationBarHeight = systemInsets.bottom;
+        additionNavigationBarHeight = hasMainTabs
+                ? dp(DialogsActivity.MAIN_TABS_HEIGHT_WITH_MARGINS)
+                : 0;
+        additionFloatingButtonOffset = hasMainTabs
+                ? dp(DialogsActivity.MAIN_TABS_HEIGHT + DialogsActivity.MAIN_TABS_MARGIN)
+                : 0;
+
+        checkUi_listViewPadding();
         checkUi_floatingButtonPosition();
         checkUi_emptyView();
 
@@ -1593,9 +1599,15 @@
     }
 
     @Override
-    public void onInsets(int left, int top, int right, int bottom) {
-        navigationBarHeight = bottom;
-        checkUi_listViewPadding();
+    public void onInsets(int left, int top, int right, int bottom) {
+        navigationBarHeight = bottom;
+        additionNavigationBarHeight = hasMainTabs
+                ? dp(DialogsActivity.MAIN_TABS_HEIGHT_WITH_MARGINS)
+                : 0;
+        additionFloatingButtonOffset = hasMainTabs
+                ? dp(DialogsActivity.MAIN_TABS_HEIGHT + DialogsActivity.MAIN_TABS_MARGIN)
+                : 0;
+        checkUi_listViewPadding();
         checkUi_floatingButtonPosition();
         checkUi_emptyView();
     }
diff --git a/TMessagesProj/src/main/java/org/telegram/ui/CallLogActivity.java b/TMessagesProj/src/main/java/org/telegram/ui/CallLogActivity.java
--- a/TMessagesProj/src/main/java/org/telegram/ui/CallLogActivity.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/CallLogActivity.java
@@ -1516,10 +1516,16 @@
 	private int navigationBarHeight;
 
 	@Override
-	public void onInsets(int left, int top, int right, int bottom) {
-		navigationBarHeight = bottom;
-
-		checkUi_listViewPadding();
+	public void onInsets(int left, int top, int right, int bottom) {
+		navigationBarHeight = bottom;
+		additionNavigationBarHeight = hasMainTabs
+				? dp(DialogsActivity.MAIN_TABS_HEIGHT_WITH_MARGINS)
+				: 0;
+		additionFloatingButtonOffset = hasMainTabs
+				? dp(DialogsActivity.MAIN_TABS_HEIGHT + DialogsActivity.MAIN_TABS_MARGIN)
+				: 0;
+
+		checkUi_listViewPadding();
 		checkUi_floatingButton();
 	}
 
